### ISSUE TITLE
eprintln

### ISSUE THREAD

**Issue by MggMuggins** [redox-os/arg-parser#5]
This is a bit of a follow-up to https://github.com/redox-os/coreutils/issues/186, I got an idea while reviewing code for error handling in userutils and redox_users:
This is a pattern that we see an AWFUL lot of:
```Rust
let user = get_user_by_id(euid).unwrap_or_else(|err| {
    eprintln!("whoami: {}", err);
    exit(1);
});
```
A situation where an error occurs and the program prints it's name, followed by a colon, followed by the error to stderr and then exits (with a nonzero exit code). 

Most (if not all) of these programs have an instance of parser in scope for the entirety of the main function, and this pattern is most often seen as a part of the main function. `ArgParser` already has access to the correct prefix for the format error, so I wonder if we could do one of the following:
1) Create a method on `ArgParser` (or a function that takes `ArgParser`, or something similar) that effectively does this given a `failure::Error`
2) Create a macro that somehow accomplishes this, given an instance of `ArgParser` and an error.

I have no experience or knowledge of writing macros, but it's a thought.

**Comment by goyox86** [comment]
I agree when I was doing the changes I saw the pattern I would add it on https://github.com/redox-os/libextra though.

**Comment by MggMuggins** [comment]
I'd love to be able to do something like this:
```Rust
eprint!(err);
```
But I'm not sure how to pull `args[0]` into that. I can write a macro like this:
```Rust
macro_rules! eprint {
    ($err:stmt) => (
        eprintln!("{:?}: {:?}", std::env::args().next(), $err);
        std::process::exit(1);
    );
    ($err:stmt, $exit_code:expr) => (
        eprintln!("{:?}: {:?}", std::env::args().next(), $err);
        std::process::exit($exit_code);
    )
}
```
I have no idea whether or not that would even work.

**Comment by MggMuggins** [comment]
Here's a functional version:
```Rust
macro_rules! fail_because {
    ($err:expr) => (
        fail_because!($err, 1);
    );
    ($err:expr, $exit_code:expr) => (
        let arg = std::env::args().next().unwrap();
        let invoc = std::path::Path::new(&arg).file_name().unwrap().to_string_lossy();
        eprintln!("{}: {}", invoc, $err);
        std::process::exit($exit_code);
    )
}
```
